Modules/Forum/classes/class.ilForum.php

```diff
@@ -1249,17 +1231,46 @@ public function getAllThreads($a_topic_id, array $params = array(), $limit = 0,
 			$this->db->setLimit($limit, $offset);
 		}
 		$res = $this->db->queryF($query, $data_types, $data);
+
+		$last_active_post_condition = "";
+		if($params['is_moderator'] == false)
+		{
+			$last_active_post_condition = " 
+				AND ( pos_status = ". $this->db->quote('1', 'integer')."
+			 	OR (pos_status = ". $this->db->quote('0', 'integer')."
+			  			AND pos_display_user_id = ". $this->db->quote($this->user->getId(), 'integer')."
+			  	))";
+		}
+
 		while($row = $this->db->fetchAssoc($res))
 		{
 			$thread = new ilForumTopic($row['thr_pk'], $params['is_moderator'], true);
 			$thread->assignData($row);
-			$threads[] = $thread;
-		}
+			$threads[$row['thr_pk']] = $thread;
 
-		return array(
-			'items' => $threads,
-			'cnt' => $cnt
-		);
+			$this->db->setLimit(1);
+			$post_res = $this->db->query('SELECT  pos_pk, pos_thr_fk, pos_date, pos_author_id, pos_display_user_id,  pos_usr_alias, import_name  
```

Comment:
This will also result in an additional `SELECT` for **every** thread iteration and **further** `SELECT` statements being executed when instantiating the `ilForumPost` with it's id, so this is counterproductive in regargs of performance.

- You'll need a sub-select or more complex `LEFT OUTER JOIN` query to solve this `most recent of group by relation` problem
- Please instantiate `ilForumPost` **without** any arguments and set the relevant data with the respective accessors afterwards.

Reply:
ok